bench: refactor random value generation

saurabhraghuvanshii opened the pull request:
Resolves #none.

## Description

> What is the purpose of this pull request?

This pull request:

-  refractor random number generation in `math/base/special/bernoulli*`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves none

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

saurabhraghuvanshii committed:
bench: random value generatio

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/betainc/benchmark/c/cephes/benchmark.c:
```diff
@@ -95,19 +95,22 @@ static double rand_double( void ) {
-	double x;
-	double a;
-	double b;
+	double x[ 100 ];
+	double a[ 100 ];
+	double b[ 100 ];
```
```suggestion
	double x[ 100 ];
	double a[ 100 ];
	double b[ 100 ];
	double elapsed;
```

Array variables should be declared before single variables.

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/betaincinv/benchmark/c/cephes/benchmark.c:
```diff
@@ -95,19 +95,23 @@ static double rand_double( void ) {
-	double x;
-	double a;
-	double b;
+	double x[ 100 ];
+	double a[ 100 ];
+	double b[ 100 ];
```
Same comment as above

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/betaincinv/benchmark/c/cephes/benchmark.c:
```diff
@@ -95,19 +95,23 @@ static double rand_double( void ) {
+	for ( i = 0; i < 100; i++ ) {
+		x[ i ] = rand_double();
+		a[ i ] = ( 1000.0*rand_double() ) + 0.1;
+		b[ i ] = ( 1000.0*rand_double() ) + 0.1;
+	}
+
+
 	t = tic();
```
```suggestion
		b[ i ] = ( 1000.0*rand_double() ) + 0.1;
	}

	t = tic();
```

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/betaln/benchmark/c/native/benchmark.c:
```diff
@@ -91,17 +91,20 @@ static double rand_double( void ) {
 	double elapsed;
-	double x;
-	double y;
+	double x[ 100 ];
+	double y[ 100 ];
```
Same comment as above

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/binomcoefln/benchmark/c/native/benchmark.c:
```diff
@@ -91,17 +91,20 @@ static double rand_double( void ) {
 	double elapsed;
-	int64_t n;
-	int64_t k;
+	int64_t n[ 100 ];
+	int64_t k[ 100 ];
 	double y;
 	double t;
```
```suggestion
	double y;
	double t;
	int64_t n[ 100 ];
	int64_t k[ 100 ];
```

Variables are listed in order of precision, with higher precision types first.

anandkaranubc commented:
@saurabhraghuvanshii Thanks for fixing these. I have left some comments.

Could you also update the title (there is a typo in 'refactor') and update the test messages to be consistent with those in previous PRs ( like https://github.com/stdlib-js/stdlib/pull/5369)?

saurabhraghuvanshii commented:
yes I can and the one I am going to ask for change to do in test file , thanks

saurabhraghuvanshii committed:
Merge branch 'stdlib-js:develop' into math-random

saurabhraghuvanshii committed:
chore: refactor file and test file

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: failed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: failed
---

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/binomcoefln/benchmark/c/native/benchmark.c:
```diff
@@ -90,18 +90,21 @@ static double rand_double( void ) {
 * @return elapsed time in seconds
 */
 static double benchmark( void ) {
+	int64_t n[ 100 ];
+	int64_t k[ 100 ];
```
These should be moved after the line `double t;`

anandkaranubc replied:
Why? [Answer](https://github.com/stdlib-js/stdlib/pull/5374#discussion_r1966561409)

saurabhraghuvanshii replied:
sorry I missed that file

saurabhraghuvanshii committed:
chore: improve order in binomcoefln/benchmark

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte commented on lib/node_modules/@stdlib/math/base/special/betaln/benchmark/c/native/benchmark.c:
```diff
@@ -90,18 +90,21 @@ static double rand_double( void ) {
 	int i;
 
+	for ( i = 0; i < 100; i++ ) {
+		x[ i ] = 1000.0 * rand_double() - 0.0;
+		y[ i ] = 1000.0 * rand_double() - 0.0;
```
```suggestion
		x[ i ] = 1000.0 * rand_double();
		y[ i ] = 1000.0 * rand_double();
```

kgryte committed:
refactor: remove noops

Signed-off-by: Athan <kgryte@gmail.com>

kgryte committed:
bench: reorder declarations

Signed-off-by: Athan <kgryte@gmail.com>

kgryte reviewed (approved):
LGTM
